h2o-core/src/main/java/water/H2O.java

```diff
@@ -734,8 +742,21 @@ else if (s.matches("decrypt_tool")) {
         i = s.incrementAndCheck(i, args);
         trgt.decrypt_tool = args[i];
       }
+      else if (s.matches("principal")) {
```

Comment:
Any reason for exposing just `principal` and not an impersonated user?

Reply:
great question - I want to avoid doing impersonation because it could be a security risk - if I make a mistake in my code there are no safeguards on the Hadoop side

for standalone deployments, it seems safer to avoid impersonation and since it is not a required use case - I am happy to skip it and risk a security issue

Reply:
np, we can add later if needed.